src/slack_api_util.ts

```diff
@@ -72,10 +72,35 @@ export async function getThreadPermalink(
     throw error;
   }
 }
-
-export async function makeFilesPublic(files: SlackFile[]): Promise<void> {
+function delay(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+export async function makeFilesPublic(files: SlackFile[]): Promise<string[]> {
+  const errors = [] as string[];
   for (const file of files) {
-    const response = await axios.post(
+    // first post a permalink in a public channel so that anyone (namely,
+    // the bot) can make the file public
+    let response = await slackAPI.post('chat.postMessage', {
+      channel: process.env.ATTACHMENTS_SLACK_CHANNEL_ID || '',
+      text: file.permalink,
+      token: process.env.SLACK_BOT_ACCESS_TOKEN,
+      unfurl_media: true,
+      unfurl_links: true,
+    });
+    if (!response.data.ok) {
+      errors.push(`Unable to post permalink to attachments channel`);
+      logger.error(
+        `SLACKAPIUTIL.makeFilesPublic: unable to post permalink to channel: ${JSON.stringify(
+          response.data
+        )}`
+      );
+      continue;
+    }
+
+    await delay(2000);
```

Comment:
Do we know why this is needed / how reliable this is? Maybe we need to wait for the message-posted even from the slack channel and then we can do this? Or maybe we have to wait until the unfurl happens (not sure if there's an event for that...). What happens if this delay is too short?

Reply:
If it doesn't work you get a red X, so there is feedback at least.  It's hard to experiment because I've only reproduced this in staging.  I think i need to set up a separate slack user locally to make it happen there.

My take is we merge this now since it gets it working, and follow up with something more elegant.

Reply:
I really wish there was a solution that didn't result in a Slackbot ping about the attachment being shared.. it's kind of annoying.